pkg/piperutils/slices.go

```diff
@@ -11,3 +13,15 @@ func ContainsInt(s []int, e int) bool {
 	}
 	return false
 }
+
+//Prefix adds a prefix to each element of the slice
+func Prefix(slice []string, prefix string) []string {
+	for idx, element := range slice {
+		element = strings.TrimSpace(element)
+		if !strings.HasPrefix(element, prefix) {
+			element = prefix + element
+		}
+		slice[idx] = element
+	}
+	return slice
```

Comment:
There are some undocumented side-effects: unless the (trimmed) element is already prefixed by the given prefix, and if a prefix is added, the element will be trimmed. I will have a look at where this is used, why this behavior is necessary at all. In any case, this is in a package and may be used from some place which is ignorant of the exact needs of the sonar command.

Reply:
I've looked at where you use that, and the side-effects seem unnecessary. Why not simply do exactly as the method is documented?

Reply:
Also, I noticed if you go with my suggestion of adding a method `appendOption()` to `sonarSettings`, then you could trim and prefix in there.

Reply:
> Why not simply do exactly as the method is documented?

I had a look at the code again and it seems that I missed to respect the `options` which can be handed in by the user.
Via that way already prefixed options could come in.

Reply:
Ok, makes sense, but this could then also be handled via `appendOption()`. I hope you don't misunderstand: I don't mind the `Prefix()` method in the utils package at all, I am just in favor of documenting its exact behavior, and nailing it tight with the unit-test. Naming it `PrefixIfNeededAndTrim()` would be another option.

Reply:
I divided the function and removed the *if needed* part.

Reply:
👍